injection/sharedmain/main.go

```diff
@@ -87,30 +103,26 @@ func MainWithContext(ctx context.Context, component string, ctors ...injection.C
 }
 
 func MainWithConfig(ctx context.Context, component string, cfg *rest.Config, ctors ...injection.ControllerConstructor) {
-	// Set up our logger.
-	loggingConfigMap, err := configmap.Load("/etc/config-logging")
-	if err != nil {
-		log.Fatal("Error loading logging configuration:", err)
-	}
-	loggingConfig, err := logging.NewConfigFromMap(loggingConfigMap)
-	if err != nil {
-		log.Fatal("Error parsing logging configuration:", err)
-	}
-	logger, atomicLevel := logging.NewLoggerFromConfig(loggingConfig, component)
-	defer flush(logger)
-	ctx = logging.WithLogger(ctx, logger)
-
-	logger.Infof("Registering %d clients", len(injection.Default.GetClients()))
-	logger.Infof("Registering %d informer factories", len(injection.Default.GetInformerFactories()))
-	logger.Infof("Registering %d informers", len(injection.Default.GetInformers()))
-	logger.Infof("Registering %d controllers", len(ctors))
+	log.Printf("Registering %d clients", len(injection.Default.GetClients()))
+	log.Printf("Registering %d informer factories", len(injection.Default.GetInformerFactories()))
+	log.Printf("Registering %d informers", len(injection.Default.GetInformers()))
+	log.Printf("Registering %d controllers", len(ctors))
```

Comment:
Any reason why we don't want to use `logger` here?

Reply:
We can't, logger creation now needs a client 😅

Reply:
heh and the clients expect a logger in the context


https://github.com/knative/pkg/blob/5112595d1ff52715e10949cb7ea24d3c31e5246e/injection/clients/kubeclient/kubeclient.go#L42-L49

Reply:
😱 that's really ugly. Since we're panicing anyway though, we might aswell not use a logger at all there? @mattmoor thoughts on that?

Reply:
An alternative route we could take: Don't try to read via API but fall back to the default config if you cannot read the file? Any preference?

Reply:
I would greatly prefer to eliminate the volume mounts and simplify the YAML.  We need API access or the watchers won't work anyways, so if there's a downside I'm missing it.

Reply:
@mattmoor the current issue is that the injection logic currently wants a proper "logger". We cannot create that without a client now though. We kinda have a circular dependency (hidden by the debouncing effect of going through context).

If you want to ditch the volume mount, let's go around and replace the logger dependency with a simple `panic` call first?

Reply:
This cannot return nil: https://github.com/knative/pkg/blob/e8df94bc5a75081fe081548c261203467448857b/logging/logger.go#L56